browser/ai_chat/tab_informer.cc

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) 2025 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "brave/browser/ai_chat/tab_informer.h"
+
+#include <utility>
+
+#include "base/strings/utf_string_conversions.h"
+#include "brave/browser/ai_chat/tab_informer_service_factory.h"
+#include "brave/components/ai_chat/core/browser/tab_informer_service.h"
+#include "brave/components/ai_chat/core/common/mojom/tab_informer.mojom.h"
+#include "chrome/browser/ui/tabs/public/tab_interface.h"
+#include "content/public/browser/navigation_entry.h"
+#include "content/public/browser/navigation_handle.h"
+#include "content/public/browser/web_contents_observer.h"
+
+namespace ai_chat {
+
+namespace {
+
+mojom::TabPtr FromWebContents(content::WebContents* web_contents) {
+  auto tab = mojom::Tab::New();
+  tab->content_id =
```

Comment:
@thypon I think we have an entry to flag `GetVisibleURL`, but I guess we're missing `GetVisibleEntry`? We should also watch out for `GetActiveEntry` since it's deprecated.

Reply:
Whoops, that was accidental - nice catch

Reply:
this one should stay unresolved because it's not specific to this PR, the comment below can be resolved for you fix